fendermint/vm/interpreter/src/fvm/state/query.rs

```diff
@@ -182,6 +182,14 @@ where
                     });
                 }
             }
+
+            // If the gas_limit is zero, set it to the block gas limit so that call will not hit
```

Comment:
Additional context for @aakoshh, when he reviews: this is what `ethers` does. Instead of estimating accurately it just uses the `BLOCK_GAS_LIMIT` and lets the validators be smart on what messages they include in their block proposal (I know we currently don't have message selection, but it seems like the most straightforward solution).

Reply:
Are you talking about `eth_call`, specifically? When I worked on the Ethereum API I saw it do the estimation if I tried to send a transaction, it's all [here](https://github.com/gakonst/ethers-rs/blob/efdc8d43318b818178c93a19a42b2b7e49e678eb/ethers-providers/src/rpc/provider.rs#L297-L323). Those use the `eth_estimateGas` and `eth_feeHistory` endpoints. I don't remember what it does for calls, but I haven't seen any block gas limit being applied.

Reply:
I think you guys misunderstood me by estimation, I'm referring more to [this](https://github.com/paradigmxyz/reth/blob/9ca44efe9f4e8768e84e855f8679e65c14a8b350/crates/rpc/rpc/src/eth/revm_utils.rs#L229) than the estimate gas.